runtime/runtime/src/gas_keys.rs

```diff
@@ -126,11 +126,12 @@ fn gas_key_storage_cost(
     let storage_config = &fee_config.storage_usage_config;
     let nonce_storage_usage = gas_key.num_nonces as u64
         * (borsh::object_length(&(0 as NonceIndex)).unwrap() as u64 + // NonceIndex is part of the key
-            borsh::object_length(&(0 as Nonce)).unwrap() as u64 + // Value of nonce
+            borsh::object_length(&Some(0 as Nonce)).unwrap() as u64 + // Value of nonce
```

Comment:
Why is this necessary? The values stored in the leaves are just nonce numbers, no? In what case can a nonce at a nonce index be a None?

Much like with values, you will need to also charge for `num_nonces` `NonceIndex`-es as there would be `num_nonces` nodes past the `public_key . gas_key` part of the trie.